data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/buffer/KafkaBuffer.java

```diff
@@ -97,4 +101,24 @@ public boolean isEmpty() {
     public Duration getDrainTimeout() {
         return drainTimeout;
     }
+
+    @Override
+    public void shutdown() {
+        shutdownInProgress.set(true);
+        executorService.shutdown();
+
+        try {
+            if (executorService.awaitTermination(EXECUTOR_SERVICE_SHUTDOWN_TIMEOUT, TimeUnit.SECONDS)) {
+                LOG.info("Successfully waited for consumer task to terminate");
+            } else {
+                LOG.warn("Consumer task did not terminate in time, forcing termination");
+                executorService.shutdownNow();
+            }
+        } catch (final InterruptedException e) {
+            LOG.error("Interrupted while waiting for consumer task to terminate", e);
+            executorService.shutdownNow();
+        }
+
+        innerBuffer.shutdown();
```

Comment:
Can we verify this line in the unit tests? I'm guessing it is not a mock due to the design, but is there another solution?

Reply:
Looks like the inner buffer is mocked, I just missed the verification here. Will add in next PR